Gemfile

```diff
@@ -1,112 +1,69 @@
 source 'https://rubygems.org'
 
-gem 'rails', '~> 4.2.7'
-
+gem 'rails',                  '5.1.2'
+gem 'simple_form',            '3.5.0'
+gem 'jquery-rails',           '4.3.1'
+gem 'turbolinks',             '5.0.1'
+gem 'sass-rails',             '5.0.6'
+gem 'coffee-rails',           '4.2.2'
+gem 'uglifier',               '3.2.0'
+gem 'jquery-ui-rails'
+gem 'bootstrap-sass',         '3.3.7'
+gem 'jquery-turbolinks'
 gem 'rack'
-gem 'ransack'
+gem 'ransack',                 '1.8.3'
 gem 'carrierwave'
-gem "pg", "~> 0.17.0"
-gem 'rmagick', '~> 2.15.4'
-gem 'simple_form'
+gem "pg"
+gem 'rmagick'
 gem 'redcarpet'
 gem 'geocoder'
 gem 'cancancan'
+gem 'paper_trail'
+gem 'therubyracer'
+gem 'twilio-ruby',             '~> 4.11.1'
 #gem "taps"
 #gem 'validates_timeliness', '~> 3.0'
-
-
-gem 'therubyracer'
-gem 'jquery-turbolinks'
+gem 'devise'
 
 source 'https://rails-assets.org' do
   gem 'rails-assets-datetimepicker'
 end
 
-# Handles text messages and voice calls
-gem 'twilio-ruby', '~> 4.11.1'
-
-#Handles authentication
-gem 'devise', '~> 3.0'
-
-# For model versioning
-gem 'paper_trail'
-
-group :test, :development do
-  gem 'thin'
-  gem "rspec-rails"
-  # gem 'rspec-activemodel-mocks' # TODO: remove if we no longer need mock_model or stub_model
-  gem "factory_girl_rails"
-  gem "capybara"
-  gem "guard-rspec", require: false
-  gem 'guard-livereload'
-  gem "mailcatcher"
-  gem "launchy"
-  gem "letter_opener"
-  gem 'better_errors'
-  gem 'binding_of_caller'
-  gem 'meta_request'
-  gem 'database_cleaner', '1.0.1'
-  gem 'test-unit'
-  gem 'poltergeist'
-  gem 'pry'
+group :production, :staging do
+  gem 'rails_12factor',       '0.0.3'
```

Comment:
Do we need this gem for Rails 5 ? I will admit I don't fully subscribe to 12 factor, especially in logging. Current production sites use file based logging and log rotation with good results. The file acts as a buffer. If someone had a large enough deployment to need a fleet of servers, they could add in a log aggregator such as logstash or fluentd.
I also am not a fan of all config in ENVVARs, but that's a tale for another day.

Reply:
@kgf Make the modifications you think necessary, I really forgot to do a gemfile cleanup.

Reply:
OK, thanks